probe-rs/src/probe/daplink/tools.rs

```diff
@@ -76,7 +76,7 @@ pub fn open_v2_device(device: Device<rusb::Context>) -> Option<DAPLinkDevice> {
     let vid = d_desc.vendor_id();
     let pid = d_desc.product_id();
     let mut handle = device.open().ok()?;
-    let language = handle.read_languages(timeout).ok()?[0];
+    let language = handle.read_languages(timeout).ok()?.get(0).cloned()?;
```

Comment:
The problem with this is, that if no serial number was given in the selector and no language descriptor is present, but the VID/PID pair would still match, this returns None, which is unfavorable because the given VID/PID matches :)

So we would have to read this only if VID/PID matches.

Reply:
I do not entirely understand what you mean. I can follow your point that the open_v2_device is only called if VID/PID were already checked (line 196). But I do not understand how to handle a missing language otherwise. The language variable is required by the read_interface_string call in line 91 to determine whether to skip the interface. Should a missing language always cause the interface to be skipped (same outcome) or should the skip check be... well skipped?
I`ll happily apply the changes, as soon I get my brain to understand what you mean :)

Reply:
Ok so I am actually in the wrong here. You are completely right that this is correct here. But my point stands for the second case I pointed this out :)